lib/perl/Genome/Model/ClinSeq/Command/UpdateAnalysis.pm

```diff
@@ -820,12 +840,32 @@ sub get_trsn{
 
   my %trsns;
   my $trsn_ref;
-
-  foreach my $instrument_data (@instrument_data){
-    my $trsn = $instrument_data->target_region_set_name;
-    if ($trsn){
-      $trsns{$trsn}=1;
-      $trsn_ref = $trsn;
+  
+  #When using validation-as-exome and my-trsn options, use user-defined target region set name (exome or validation)
+  if ($self->validation_as_exome && $self->my_trsn) {
```

Comment:
I haven't tested it, so you'll likely want to double check my work, but I think now that lines 840-870 can be condensed down into something like this (getting rid of the extra `for` loop and some of the nesting):

````perl
my %trsns;
my $trsn_ref;
my $user_trsn;

if ($self->validation_as_exome && $self->my_trsn) {
    $user_trsn = $self->my_trsn;
    $self->warning_message("Using '$user_trsn' target region specified by my-trsn.");
}

for my $instrument_data (@instrument_data){
    if (my $trsn =  $instrument_data->target_region_set_name) {
        $trsns{$trsn} = 1;
        if ($user_trsn && Genome::FeatureList->get(name => $trsn, content_type => $user_trsn)) {
            $trsn_ref = $trsn;
        } else {
            $trsn_ref = $trsn;
        }
    }
}
````

A quick thought as well: this will blow up (both my implementation and the already existing one) if there is more than one feature list with the same trsn and content type. I assume that shouldn't happen? It looks like this logic exists to basically make sure that a feature list with the user specified content type and trsn exists before continuing, but we don't actually use the feature list at this point in the code, is that correct?